docs/proposals/operator-hub.md

```diff
@@ -0,0 +1,197 @@
+# Operator Hub integration
+odo currently supports creation, listing and deletion of services created from
+the Service Catalog. With newer versions of OCP, Service Catalog is being
+deprecated in favour of Operator Hub.
+
+This proposal discusses adding similar support for Operator Hub as is currently
+available for Service Catalog. Idea is to be able to create, list and delete
+the services (basically Custom Resources in Operator framework terminology) on
+an OpenShift or Kubernetes cluster. Considering the ongoing `kclient` and DevFile
+efforts, we intend to add support for Kubernetes from day 1. However, there
+will be certain prerequisites that need to be satisfied in case of Kubernetes.
+
+Table of Contents
+- [Which Operators do we support](#which-operators-do-we-support)
+- [Installing Operators](#installing-operators)
+- [List the services/operators](#list-the-servicesoperators)
+  - [Both Service Catalog and Operator Hub are enabled](#both-service-catalog-and-operator-hub-are-enabled)
+  - [Only Service Catalog is enabled](#only-service-catalog-is-enabled)
+  - [Only Operator Hub is enabled](#only-operator-hub-is-enabled)
+  - [Kubernetes](#kubernetes)
+- [Link components with operands](#link-components-with-operands)
+- [Listing the services](#listing-the-services)
+- [Describe deployed services](#describe-deployed-services)
+- [Delete the service](#delete-the-service)
+- [Testing on Kubernetes](#testing-on-Kubernetes)
+- [Questions](#questions)
+- [GitHub issues related to this task](#github-issues-related-to-this-task)
+
+## Which Operators do we support
+Operator Hub is not the only place to find/list Operators that could be
+installed on a OpenShift/k8s cluster. There’s also KUDO. However, to start
+things off, we must focus on a limited number of Operators and iterate from
+there. Hence, we will focus only on OperatorHub (kind of obvious 😉).
+
+There are three main categories of Operators as described in the OpenShift
+docs. We could focus on one of those categories or select specific Operators
+from each of the categories.
+
+From a development perspective, we might just take a CRD definition and throw
+it to the Kubernetes API. The controller running in background would take care
+of spinning up the service. So, it wouldn’t matter a lot to differentiate
+between what we support and what we don’t. However, limiting our scope to a few
+Operators will keep us from being overwhelmed.
+
+## Installing Operators
+There’s an open issue about adding capability in odo to install Operators and
+installing “Service Binding Operator” (which helps bind applications with
+Operator-backed services) as a one-off command.
+
+As far as creating services from Service Catalog is concerned, we expect the
```

Comment:
I would explicitly say that odo will just  "instantiate" operands and that it is expected that someone with the cluster-admin will enable (install) operators first.

Reply:
+1

Reply:
Agree with @kadel an admin needs to do something outside of `odo` to make the operator/CRD available.

Reply:
I agree with your concern @kadel & @sspeiche, but if you read the paragraph fully, you'll see that your concern is exactly what is proposed. :wink: There are more words because we have #2464 which talks about letting developers do "few" admin tasks but I don't agree with that notion/idea. Let me know if this is still unclear/vague/lengthy and I'll change it.